CHANGELOG.md

```diff
@@ -6,6 +6,64 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html),
 and is generated by [Changie](https://github.com/miniscruff/changie).
 
+## v0.16.0 - 2025-02-19
+
+### 🔥 Breaking Changes
+- To match automatic configuration, `insecure-entitlements` now includes `security.insecure` when configuring the engine manually by @jedevc in https://github.com/dagger/dagger/pull/9513
+- Module load performance is improved and related API refactored by @sipsma in https://github.com/dagger/dagger/pull/9505 \
+  Loading of modules (the `load module` step shown in progress output) is faster now in many cases. In particular: 
+  - Cache utilization of module loading is greatly improved, which can decrease load times by up to a factor of 10 when re-calling functions after changing source code in a Daggerized repo. 
+  - Less extraneous files are loaded from the source repository
+  
+  Users of modules with large numbers of dependencies or in large git repositories are expected to see the most immediate benefit.
+  
+  For some concrete numbers, here are `load module` times for the `dagger-dev`'s module in Dagger's repository under different scenarios:
+  - `dagger call --help` on new engines with an empty cache
+    - v0.15.4: 1m20s
+    - v0.16.0: 1m1s
+    - ~23% faster
+  - re-running `dagger call --help` with no file changes in the repo:
+    - v0.15.4: 10.9s
+    - v0.16.0: 2.8s
+    - ~74% faster
+  - re-running `dagger call --help` after making a change to a random source code file in the repo:
+    - v0.15.4: 32.1s
+    - v0.16.0: 2.8s
+    - ~91% faster
+  
+  These improvements in cache utilization are also setup for future improvements not only in `load module` times but function call times more generally.
+  
+  This improvement comes with a few breaking changes that are not expected to impact most users. However, the changes require that users upgrade their CLI to v0.16.0 in order to connect to v0.16.0 engines. Older CLIs will error when connecting to newer engines.
+  
+  In terms of the breaking changes, impacted users are those who:
+  - have function code that imports local libraries outside of the dagger module source directory
+  - rely on a package manager configuration file (e.g. go.mod) that exists in a parent directory of the dagger module source directory
+  
+  They may need to update their dagger.json configuration file to add an explicit `"include"` setting to ensure those extra files or directories are loaded with the module. This is because numerous files previously implicitly loaded are now skipped by default, which contributes to the performance improvements.
+  
+  For example, if your module code is in `.dagger` and relies on the `go.mod` file in the parent directory, you would add the following to your `dagger.json` file:
```

Comment:
```suggestion
  For [example](https://github.com/dagger/dagger/pull/9526), if your module code is in `.dagger` and relies on the `go.mod` file in the parent directory, you would add the following to your `dagger.json` file:
```

Reply:
That PR diff is a little different than the `go.mod` example mentioned here. Based on my daggerverse crawl very very few users are impacted and those that are all involved go.mod, so I'll just leave this as is for now